File: implementations/go/examples/internal/application/command/create_order_handler.go
package command

import (
	"context"

	"github.com/example/order-service/internal/domain/order"
)

type CreateOrderCommand struct {
	UserID string
	Items  []order.ItemInput
}

type CreateOrderHandler struct {
	repo order.Repository
}

func NewCreateOrderHandler(repo order.Repository) *CreateOrderHandler {
	return &CreateOrderHandler{repo: repo}
}

func (h *CreateOrderHandler) Handle(ctx context.Context, cmd CreateOrderCommand) error {
	o, err := order.New(cmd.UserID, cmd.Items)
	if err != nil {
		return err
	}
	return h.repo.Save(ctx, o)
}

File: implementations/fastapi/examples/src/order/application/command/create_order_handler.py
from dataclasses import dataclass

from ...domain.order import ItemInput, Order
from ...domain.repository import OrderRepository


@dataclass
class CreateOrderCommand:
    user_id: str
    items: list[ItemInput]


class CreateOrderHandler:

    def __init__(self, repo: OrderRepository) -> None:
        self._repo = repo

    async def execute(self, cmd: CreateOrderCommand) -> None:
        order = Order.create(cmd.user_id, cmd.items)
        await self._repo.save(order)
